### PR TITLE
Authorizations - Expired auth should mark order as failed

### PR DESCRIPTION
**Pull request by nagpai**
Related to 
https://github.com/Automattic/woocommerce-payments-server/issues/3254

#### Changes proposed in this Pull Request
Authorization payment intent expiry should end up changing order status to `failed` . Currently it is changed to `canceled`, This will help the user to reattempt payment. 

#### Testing instructions

* Existing unit tests should pass

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6121` or branch name `update/improve-authorization-expiry` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=update/improve-authorization-expiry) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 35217716552a84493303d62083feee072e5ba4d4
- Build time: 2023-07-13 10:57:05 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/class-wc-payments-order-service.php [hunk 1]
```diff
@@ -216,7 +216,7 @@ public function mark_payment_capture_failed( $order, $intent_id, $intent_status,
 	}
 
 	/**
-	 * Updates an order to cancelled status, while adding a note with a link to the transaction.
+	 * Update an order to failed status, and add note with a link to the transaction. Context - when a Payment Intent expires.
```
Comment (kalessil) [inline includes/class-wc-payments-order-service.php]: I suggest mentioning that failed statutes enable the buyer to attempt a new payment.

File: tests/unit/test-class-wc-payment-gateway-wcpay.php [hunk 2]
```diff
@@ -1269,7 +1269,7 @@ public function test_capture_charge_expired() {
 		);
 		$this->assertStringContainsString( 'expired', $note->content );
 		$this->assertEquals( Payment_Intent_Status::CANCELED, $order->get_meta( '_intention_status', true ) );
-		$this->assertEquals( Order_Status::CANCELLED, $order->get_status() );
+		$this->assertEquals( Order_Status::FAILED, $order->get_status() );
```
Comment (kalessil) [inline tests/unit/test-class-wc-payment-gateway-wcpay.php]: Strict assertions, please.

File: tests/unit/test-class-wc-payments-order-service.php [hunk 3]
```diff
@@ -697,14 +697,14 @@ public function test_mark_payment_capture_failed_null_intent_status() {
 	public function test_mark_payment_capture_expired_with_provider( $fraud_outcome, $expected_fraud_outcome, $expected_fraud_meta_box ) {
 		// Arrange: Create the intent, get the proper order status variations, and get the charge. Set the fraud outcome status.
 		$intent            = WC_Helper_Intention::create_intention( [ 'status' => Payment_Intent_Status::CANCELED ] ); // Stripe uses single 'l'.
```
Comment (kalessil) [inline tests/unit/test-class-wc-payments-order-service.php]: The comment seems to be not relevant anymore, or that's just me?

File: tests/unit/test-class-wc-payments-order-service.php [hunk 4]
```diff
@@ -697,14 +697,14 @@ public function test_mark_payment_capture_failed_null_intent_status() {
...
-		$order_status      = Order_Status::CANCELLED; // WCPay uses double 'l'.
+		$order_status      = Order_Status::FAILED;
 		$wc_order_statuses = wc_get_order_statuses(); // WooCommerce uses single 'l' for US English.
```
Comment (kalessil) [inline tests/unit/test-class-wc-payments-order-service.php]: The comment seems to be not relevant anymore, or that's just me?

**Review by kalessil — CHANGES_REQUESTED** [review]

File: tests/unit/test-class-wc-payments-order-service.php (hunk 3 above)
Comment (nagpai) [reply tests/unit/test-class-wc-payments-order-service.php:699]: No, it's me :) . Thanks for the callout.  Notes to myself - Watch the comments around the code thats's changed!

File: includes/class-wc-payments-order-service.php (hunk 1 above)
Comment (nagpai) [reply includes/class-wc-payments-order-service.php]: Thanks. I have added a comment now 230c22f

File: tests/unit/test-class-wc-payment-gateway-wcpay.php (hunk 2 above)
Comment (nagpai) [reply tests/unit/test-class-wc-payment-gateway-wcpay.php]: Changed this to assertSame

File: tests/unit/test-class-wc-payments-order-service.php (hunk 3 above)
Comment (nagpai) [reply tests/unit/test-class-wc-payments-order-service.php]: Thanks for spotting. I have removed it.

File: tests/unit/test-class-wc-payments-order-service.php (hunk 4 above)
Comment (nagpai) [reply tests/unit/test-class-wc-payments-order-service.php]: Sorry this remained from earlier change. I have now removed it

File: includes/class-wc-payments-order-service.php [hunk 5]
```diff
@@ -216,7 +216,7 @@ public function mark_payment_capture_failed( $order, $intent_id, $intent_status,
 	}
 
 	/**
-	 * Updates an order to cancelled status, while adding a note with a link to the transaction.
+	 * Update an order to failed status, and add note with a link to the transaction. Context - when a Payment Intent expires. Changing the status to failed will enable the buyer to re-attempt payment.
```
Comment (dmvrtx) [inline includes/class-wc-payments-order-service.php]:
```suggestion
	 * When a payment intent expires update an order to failed status, and add note with a link to the transaction.
	 *
	 * Context: changing the status to failed will enable the buyer to re-attempt payment.
```

Docblock is expected to have a brief summary on the first line follow-ed up by a detailed overview if needed.

**Review by dmvrtx — APPROVED** [review]
Provisionally approve based on code review.

**Review by KarlisJ — APPROVED** [review]
LGTM, approve given you implement the suggestion from Valery

File: includes/class-wc-payments-order-service.php (hunk 5 above)
Comment (nagpai) [reply includes/class-wc-payments-order-service.php]: Thanks Valery!. I have changed it here - b4a4c65

**Review by kalessil — APPROVED** [review]

**Pull request merged**